addons/udes_security/models/models.py

```diff
@@ -74,6 +93,15 @@ def _set_fields_readonly(doc, res):
 
             res["arch"] = etree.tostring(doc)
             # Remove archive option in tree view and make calendar readonly
-            res["readonly_desktop_user"] = True
+            res["readonly_desktop_user"] = True  # Used in basic_view.js
+
+        can_archive_record = ResArchivingRestrictions._get_can_archive(model)
+        if not can_archive_record:
+            doc = etree.XML(res["arch"])
+            if doc.tag == "form":
```

Comment:
Q: What about the archive option from the tree view when you select some records?

Reply:
That is handled by `cannot_archive_record` that's shoved into res, and picked up in BasicView.js